packages/block/src/header.ts

```diff
@@ -219,8 +219,15 @@ export class BlockHeader {
       }
     }
 
-    if (options.hardforkByBlockNumber) {
-      this._common.setHardforkByBlockNumber(number.toNumber())
+    if (options.hardforkByBlockNumber !== undefined && options.hardforkByTD !== undefined) {
+      throw new Error(
+        `The hardforkByBlockNumber and hardforkByTD options can't be used in conjunction`
+      )
+    }
+
+    const hardforkByBlockNumber = options.hardforkByBlockNumber ?? false
+    if (hardforkByBlockNumber || options.hardforkByTD) {
+      this._common.setHardforkByBlockNumber(number.toNumber(), options.hardforkByTD)
```

Comment:
we don't need `number.toNumber()` here anymore since we expanded the input params for `setHardforkByBlockNumber` it accepts BN now.

```suggestion
      this._common.setHardforkByBlockNumber(number, options.hardforkByTD)
```